go/chat/storage/storage.go

```diff
@@ -897,7 +897,14 @@ func (s *Storage) FetchMessages(ctx context.Context, convID chat1.ConversationID
 			}
 		}
 		sres = rc.Result()
-		res = append(res, &sres[0])
+		msg := &sres[0]
+
+		// If we have a versioning error but our client now understands the new
+		// version, don't return the error message
+		if msg.IsError() && msg.Error().ParseableVersion() {
```

Comment:
Think we should just stored this on the error like you said.

Reply:
we have to leave the parsing in for a few releases otherwise old clients wont' be able to refetch messages which was what the hope was for introducing V4 with reputability for ephemeral emssages in the next release